fmralign/alignment_methods.py

```diff
@@ -465,3 +467,239 @@ def fit(self, X, Y):
+    Method of alignment based on the Individualized Neural Tuning model, by Feilong Ma et al. (2023).
+    It works on 4D fMRI data, and is based on the assumption that the neural response to a stimulus is shared across subjects.
+    It uses searchlight/parcelation alignment to denoise the data, and then computes the stimulus response matrix.
+    See article : https://doi.org/10.1162/imag_a_00032
+    """
+
+    def __init__(
+        self,
+        template="pca",
+        decomp_method=None,
+        n_components=None,
+        alignment_method="searchlight",
+        n_jobs=1,
+    ):
+        """
+        Initialize the IndividualizedNeuralTuning object.
+        Parameters:
+        --------
+
+        - template (str): The type of template to use for alignment. Default is "pca".
+        - decomp_method (str): The decomposition method to use. Default is None.
+        - alignment_method (str): The alignment method to use. Can be either "searchlight" or "parcelation", Default is "searchlight".
+        - n_components (int): The number of latent dimensions to use in the shared stimulus information matrix. Default is None.
+        - n_jobs (int): The number of parallel jobs to run. Default is -1.
+
+        Returns:
+        --------
+        None
+        """
+
+        self.n_s = None
+        self.n_t = None
+        self.n_v = None
+        self.labels = None
+        self.alphas = None
+        self.alignment_method = alignment_method
+        if alignment_method == "parcelation":
+            self.parcels = None
+
+        elif (
+            alignment_method == "searchlight"
+            or alignment_method == "ensemble_searchlight"
+        ):
+            self.searchlights = None
+            self.distances = None
+            self.radius = None
+
+        self.tuning_data = []
+        self.denoised_signal = []
+        self.decomp_method = decomp_method
+        self.template = template
+        self.n_components = n_components
+        self.n_jobs = n_jobs
+
+    #######################################################################################
+    # Computing decomposition
+
+    @staticmethod
+    def _tuning_estimator(shared_response, target):
+        """
+        Estimate the tuning weights for individualized neural tuning.
+
+        Parameters:
+        --------
+            - shared_response (array-like):
+                The shared response matrix.
+            - target (array-like):
+                The target matrix.
+            - latent_dim (int, optional):
+                The number of latent dimensions. Defaults to None.
+
+        Returns:
+        --------
+            array-like: The estimated tuning weights.
+
+        """
+        if shared_response.shape[1] < shared_response.shape[0]:
+            return (np.linalg.pinv(shared_response)).dot(target)
+        return np.linalg.inv(shared_response).dot(target)
+
+    @staticmethod
+    def _stimulus_estimator(full_signal, n_t, n_s, latent_dim=None):
```

Comment:
it uses nothing from self, hence  it should be a function ?
Why not use self.n_t, self.n_s instead of giving them as arguments ?

Reply:
You advised me before not to give it self agrs and make it its own function, but for me a static method is easier to access it outside of the model (for experiments for example)